src/common/io/tls/session.c

```diff
@@ -144,77 +184,33 @@ tlsSessionRead(THIS_VOID, Buffer *buffer, bool block)
 
     ssize_t result = 0;
 
+    // If no TLS data pending then check the socket to reduce blocking
+    if (!SSL_pending(this->session))
+        sckSessionReadyRead(this->socketSession);
+
     // If blocking read keep reading until buffer is full
     do
     {
-        // If no tls data pending then check the socket
-        if (!SSL_pending(this->session))
```

Comment:
Should we really move this out of the loop? Does postgres do it outside the loop?

Reply:
Postgres has it in the loop and so should we.